pkg/controller/periodic/periodic.go

```diff
@@ -140,3 +147,37 @@ func (c *Controller) processNextWorkItem() bool {
 
 	return true
 }
+
+func (c *Controller) updateConfig() bool {
+	cfg := c.configurator.Config()
+	if c.interval != cfg.Interval {
+		c.interval = cfg.Interval
+		klog.V(4).Infof("Gathering interval set to %s", cfg.Interval)
+		return true
+	} else {
+		return false
+	}
+}
+
+// Runs loop for checking on configuration change to check for periodic-related changes.
+// Returns a channel that it writes to, when the config change influenced the controller.
+func (c *Controller) watchConfig(stopCh <-chan struct{}) chan struct{} {
+	configCh := make(chan struct{}, 1)
+	globalConfigCh, cancelFn := c.configurator.ConfigChanged()
+
+	watchChanges := func() {
```

Comment:
nit: `go func() { }`

(you don't need this func stored, right?)

Reply:
I know this, but want to name the func to describe what's it's purpose. Would like to keep it that way if you don't mind